### ISSUE TITLE
(aws-cdk):  default typescript `cdk init` swallows stack traces 

### ISSUE THREAD

**Issue by blacha** [aws/aws-cdk#15897]
after using a `cdk init --language=typescript`

The default setup swallows stack traces

### Reproduction Steps
```
cdk init --language=typescript
echo "throw new Error('foo')" > bin/lib.ts
npx cdk synth
```

### What did you expect to happen?

I would expect to see the full stack trace as if i had run it with `node`

```
node bin/lib.js 
/home/blacha/tmp/cdk-init/lib/bin/lib.ts:1
throw new Error('foo')
      ^

Error: foo
    at Object.<anonymous> (/home/blacha/tmp/cdk-init/lib/bin/lib.ts:1:7)
    at Module._compile (internal/modules/cjs/loader.js:1072:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1101:10)
    at Module.load (internal/modules/cjs/loader.js:937:32)
    at Function.Module._load (internal/modules/cjs/loader.js:778:12)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:76:12)
    at internal/main/run_main_module.js:17:47
```

### What actually happened?

The stack trace is swallowed

```
foo
Subprocess exited with error 1
```

### Environment

  - **CDK CLI Version  :** 1.116.0 (build d04661d)
  - **Framework Version:** 1.116.0 (build d04661d)
  - **Node.js Version:** v14.17.4
  - **OS               :** Ubuntu 21
  - **Language (Version):** 

### Other

This is caused by `npx` and `ts-node` which is setup https://github.com/aws/aws-cdk/blob/master/packages/aws-cdk/lib/init-templates/v1/app/typescript/cdk.template.json

https://github.com/TypeStrong/ts-node/issues/592

Could the defaults be changed to not use a `npx ts-node` combination? 

--- 

This is :bug: Bug Report

**Comment by rix0rrr** [comment]
I suppose you can replace `npx ts-node` with `node_modules/.bin/ts-node`, but I don't know if that's safe.

**Comment by blacha** [comment]
Upstream would be my preferred place for this to be fixed too..

After some more digging, it looks like it may be fixed with the latest npm v7.20.x

Using the latest LTS node
```
$ node --version
v14.17.4
$ npm --version
6.14.14
$ npx ts-node bin/lib.ts
foo
```

Using the latest version of node
```
$ node --version
v16.6.1
$ npm --version
7.20.3
$ npx ts-node lib/bin.ts

/home/blacha/tmp/cdk-init/lib/bin/lib.ts:1
throw new Error('foo')
      ^
Error: foo
    at Object.<anonymous> (/home/blacha/tmp/cdk-init/lib/lib.ts:1:7)
    ...
```

Using the LTS node with the latest npm i could install with `npm i -g npm`

```
$ node --version
v14.17.4
$ npm --version
7.20.5
$ npx ts-node lib/bin.ts

/home/blacha/tmp/cdk-init/lib/bin/lib.ts:1
throw new Error('foo')
      ^
Error: foo
    at Object.<anonymous> (/home/blacha/tmp/cdk-init/lib/lib.ts:1:7)
    ...
```

So maybe we can close this and just tell people to upgrade their NPM!

**Comment by peterwoodworth** [comment]
Seems like that's the most appropriate solution here. Closing now
